pkg/monitor/cluster/nodeconditions_test.go

```diff
@@ -45,44 +61,204 @@ func TestEmitNodeConditions(t *testing.T) {
+		m:      m,
 	}
 
 	m.EXPECT().EmitGauge("node.count", int64(2), map[string]string{})
 	m.EXPECT().EmitGauge("node.conditions", int64(1), map[string]string{
-		"nodeName": "aro-master-0",
-		"status":   "True",
-		"type":     "MemoryPressure",
+		"nodeName":     "aro-master-0",
+		"status":       "True",
+		"type":         "MemoryPressure",
+		"spotInstance": "false",
 	})
 	m.EXPECT().EmitGauge("node.conditions", int64(1), map[string]string{
-		"nodeName": "aro-master-1",
-		"status":   "False",
-		"type":     "Ready",
+		"nodeName":     "aro-master-1",
+		"status":       "False",
+		"type":         "Ready",
+		"spotInstance": "false",
 	})
 
 	m.EXPECT().EmitGauge("node.kubelet.version", int64(1), map[string]string{
 		"nodeName":       "aro-master-0",
-		"kubeletVersion": "v1.17.1+9d33dd3",
+		"kubeletVersion": kubeletVersion,
 	})
 	m.EXPECT().EmitGauge("node.kubelet.version", int64(1), map[string]string{
 		"nodeName":       "aro-master-1",
-		"kubeletVersion": "v1.17.1+9d33dd3",
+		"kubeletVersion": kubeletVersion,
 	})
 
-	err := mon.emitNodeConditions(ctx)
+	err = mon.emitNodeConditions(ctx)
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestGetSpotInstances(t *testing.T) {
+	ctx := context.Background()
+
+	spotProvSpec, err := json.Marshal(azureproviderv1beta1.AzureMachineProviderSpec{
+		SpotVMOptions: &azureproviderv1beta1.SpotVMOptions{},
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	provSpec, err := json.Marshal(azureproviderv1beta1.AzureMachineProviderSpec{})
 	if err != nil {
 		t.Fatal(err)
 	}
+
+	for _, tt := range []struct {
+		name                 string
+		maocli               maoclient.Interface
+		node                 corev1.Node
+		expectedSpotInstance bool
+	}{
+		{
+			name: "node is a spot instance",
+			maocli: maofake.NewSimpleClientset(&machinev1beta1.Machine{
+				Spec: machinev1beta1.MachineSpec{
+					ProviderSpec: machinev1beta1.ProviderSpec{
+						Value: &kruntime.RawExtension{
+							Raw: spotProvSpec,
+						},
+					},
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "aro-spot-0",
+					Namespace: "openshift-machine-api",
+				},
+			}),
+			node: corev1.Node{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:        "aro-spot-0",
+					Annotations: map[string]string{"machine.openshift.io/machine": "openshift-machine-api/spot-0"},
+				},
+			},
+			expectedSpotInstance: true,
+		},
+		{
+			name: "node is not a spot instance",
+			maocli: maofake.NewSimpleClientset(&machinev1beta1.Machine{
+				Spec: machinev1beta1.MachineSpec{
+					ProviderSpec: machinev1beta1.ProviderSpec{
+						Value: &kruntime.RawExtension{
+							Raw: provSpec,
+						},
+					},
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "master-0",
+					Namespace: "openshift-machine-api",
+				},
+			}),
+			node: corev1.Node{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:        "aro-master-0",
+					Annotations: map[string]string{"machine.openshift.io/machine": "openshift-machine-api/master-0"},
```

Comment:
nit: to improve readability could you put each key on its own line? (we also do this elsewhere in our tests)